src/lib.rs

```diff
@@ -146,6 +148,28 @@ impl ToolFamily {
             ToolFamily::Clang => "-E",
         }
     }
+
+    /// What the flags to enable all warnings
+    fn warnings_flags(&self) -> &'static [&'static str] {
+        static MSVC_FLAGS: &'static [&'static str] = &["/Wall"];
+        static GNU_FLAGS: &'static [&'static str] = &["-Wall", "-Wpedantic", "-Wextra"];
+        static CLANG_FLAGS: &'static [&'static str] = &["-Weverything"];
+
+        match *self {
+            ToolFamily::Msvc => &MSVC_FLAGS,
+            ToolFamily::Gnu => &GNU_FLAGS,
+            ToolFamily::Clang => &CLANG_FLAGS,
+        }
+    }
+
+    /// What the flag to turn warning into errors
+    fn warnings_to_errors_flag(&self) -> &'static str {
+        match *self {
+            ToolFamily::Msvc => "/WX",
+            ToolFamily::Gnu |
+            ToolFamily::Clang => "-Werror"
+        }
```

Comment:
I think turning warnings into errors only make sense if you are a developer of the crate using gcc-rs (i.e., enabling the warnings as errors only in a "developer mode" or something). Any user of this crate, or any other crate depending on it, could fail during compile time. Some warnings only appear on some architecture. Some people might have a version of the compiler that have a new warning, but not the developers. Hence, if the developers use another architecture or another compiler they will not have any warning, thus no error. But another person might have the error.

I think it would make sense to put some comment on the risk of using this function in your `build.rs`?

What do you think?